more_itertools/more.py

```diff
@@ -4726,3 +4729,53 @@ def join_mappings(**field_to_map):
             ret[key][field_name] = value
 
     return dict(ret)
+
+
+def _complex_sumprod(v1, v2):
```

Comment:
Since `_sumprod` is Python 3.12+, I'm using the backup implementation from `recipes` for older versions.

Reply:
So that the high-precision calculation doesn't get wasted, I recommend an alternate `_sumprod` implemented as `math.fsum(map(operator.mul, v1, v2))`

Reply:
Added in acdbb80fe181406eab4466db8087019d862a0f46